File: direct_native/example/todo_app/android/app/src/main/java/com/example/directnativeandroid/DNBridge.kt
package com.example.directnative

import android.app.Activity
import org.json.JSONObject

class DNBridge(private val activity: Activity) {
    private val renderer = DNRenderer(activity)

    external fun initialize()
    
    fun nativeRender(uiDescription: String) {
        val jsonDescription = JSONObject(uiDescription)
        activity.runOnUiThread {
            renderer.render(jsonDescription)
        }
    }

    companion object {
        init {
            System.loadLibrary("direct_native")
        }
    }
}

File: direct_native/example/todo_app/ios/DirectNativeIOS/DirectNative/DNBridge.swift
import SwiftUI

class DNBridge: ObservableObject {
    @Published var rootView: AnyView = AnyView(EmptyView())
    private let renderer = DNRenderer()
    
    func initialize() {
        // Perform any necessary initialization
        print("DNBridge initialized")
    }
    
    func nativeRender(_ uiDescription: String) {
        if let data = uiDescription.data(using: .utf8),
           let json = try? JSONSerialization.jsonObject(with: data, options: []) as? [String: Any] {
            DispatchQueue.main.async {
                self.rootView = self.renderer.render(viewDescription: json)
            }
        }
    }
}
